### PR TITLE
method based disease setup

### PR DESCRIPTION
**Pull request by hussain-jafari**
### Description
- *Category*: feature
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-4613

### Changes and notes
Perform BaseDiseaseState setup using more methods.

### Testing
Ran MNCH child sim for a few time steps.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
         long_description = f.read()
 
     install_requirements = [
-        "vivarium>=2.0.0",
+        "vivarium>=2.1.0",
```
Comment (rmudambi) [inline setup.py:18]: Does this require vivarium 2.1.0 (i.e. is there some reason to believe it wouldn't be compatible with vivarium 2.0.1)? If not, we shouldn't touch this pin.

**Review by rmudambi — APPROVED** [review]

**Review by mattkappel — COMMENTED** [review]
From what I can tell, this is mostly a refactor. Is that right? The changelog "Use methods to setup BaseDiseaseState" is a little vague.

**Review by mattkappel — APPROVED** [review]
Suggest adding more clarity to the changelog description

**Review by patricktnast — APPROVED** [review]

**Comment by hussain-jafari** [comment]
@mattkappel How about something like "Perform actions in DiseaseState setup using class methods rather than hardcoding to allow for cleaner subclassing"?

**Pull request merged**